ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -751,8 +751,11 @@ def _compile_pyarrow_udf(self, udf_node: ops.ScalarUDF) -> str:
 
     def _register_udfs(self, expr: ir.Expr) -> None:
         with self.begin() as con:
-            for sql in self._compile_udfs(expr):
-                con.exec_driver_sql(sql)
+            for register_udf in self._compile_udfs(expr):
```

Comment:
`_compile_udfs` is called in two places currently:

- `_register_udfs`, where the strings/functions yielded by `_compile_udfs` are immediately executed
- `Backend.compile`, where it's only needed for the side effect of registering the dynamically generated UDF ops with the backend (so they're compilable).

If the latter use case was extracted out to a new generic function, then `_compile_udfs` and `_register_udfs` could be merged into the same method, which IMO would feel a bit cleaner than having `_compile_udfs` yield strings or functions to immediately execute on the connection.

Reply:
Done!